### ISSUE TITLE
Advice request - Training agents to find each other in a maze

### ISSUE THREAD

**Issue by tkggwatson** [Unity-Technologies/ml-agents#1796]
Hi,

I'm wondering if anyone could give me some advice please? My setting is a maze where the agents need to find each other. The agents can spawn anywhere in the maze, and have forwards and rotation controls and a set of visual perception rays firing at various angles in front of them.

I've got this working and the problem is successfully being solved, but as I'm new to this I'd like to hear how others might do it, and especially ways in which I could speed up training (as it takes about 2 hours and 200k episodes currently).

My current training set up is 2 agents per maze, 15 separate mazes. Discrete action space with 3 stacked observation vectors. The agents receive a -1/maxStep time penalty and a +2 reward on locating the other agent and moving to within a few metres of it. I've also given them a tiny penalty for colliding with the walls, which gives a slightly more pleasing visual quality.

I have no idea about setting hyperparameters, so I've used the defaults from the demos. Advice here would be great as I have little experience tuning them. Also any advice on my brain configuration or training area setup would be greatly appreciated.

One issue I have is that the first ~80k training episodes, the agents prefer to just spin in circles instead of explore. Any advice on how to get them over that hurdle quicker please?

Many many thanks in advance for your feedback!

Regards
Graham

**Comment by MarcoMeter** [comment]
> One issue I have is that the first ~80k training episodes, the agents prefer to just spin in circles instead of explore. Any advice on how to get them over that hurdle quicker please?

You could enable curiosity.

**Comment by xiaomaogy** [comment]
Thanks for reaching out to us. Hopefully you were able to resolve your issue.  We are closing this due to inactivity, but if you need additional assistance, feel free to reopen the issue.
